semantickernel-experimental/src/main/java/com/microsoft/semantickernel/connectors/data/redis/RedisHashSetVectorStoreRecordMapper.java

```diff
@@ -91,28 +111,35 @@ public RedisHashSetVectorStoreRecordMapper<Record> build() {
                 throw new SKException("vectorStoreRecordDefinition is required");
             }
 
-            ObjectMapper mapper = new ObjectMapper();
-
             return new RedisHashSetVectorStoreRecordMapper<>(record -> {
                 try {
-                    ObjectNode jsonNode = mapper.valueToTree(record);
+                    ObjectNode jsonNode = objectMapper.valueToTree(record);
+
                     String key = jsonNode
                         .get(recordDefinition.getKeyField().getEffectiveStorageName()).asText();
                     jsonNode.remove(recordDefinition.getKeyField().getEffectiveStorageName());
 
-                    Map<String, String> resultMap = new HashMap<>();
-                    Iterator<Entry<String, JsonNode>> fields = jsonNode.fields();
-                    while (fields.hasNext()) {
-                        Map.Entry<String, JsonNode> field = fields.next();
-                        if (field.getValue().isTextual()) {
-                            resultMap.put(field.getKey(), field.getValue().asText());
-                        } else {
-                            resultMap.put(field.getKey(),
-                                mapper.valueToTree(field.getValue()).toString());
+                    Map<byte[], byte[]> storage = new HashMap<>();
+                    for (VectorStoreRecordDataField field : recordDefinition.getDataFields()) {
+                        JsonNode value = jsonNode.get(field.getEffectiveStorageName());
+                        if (value != null) {
+                            storage.put(
+                                field.getEffectiveStorageName().getBytes(StandardCharsets.UTF_8),
```

Comment:
Just a comment. I'm not expecting you to change any code - unless you really want to ;^). You have that nice, little "byte[] stringToBytes(String s)" method elsewhere. Maybe something that could be reused?